imap_processing/cdf/utils.py

```diff
@@ -16,37 +16,36 @@
 logger = logging.getLogger(__name__)
 
 
-def calc_start_time(
-    shcoarse_time: float,
+def convert_met_to_datetime64(
```

Comment:
I don't have a strong opinion on this, but wondering if we should instead call this `get_met()` or `convert_to_mission_met()` or something similar? Because we are not simply converting times here, we are returning times that are converted _and_ are now relative to the `launch_time`.

Reply:
I think that means we need a new name then if you're confused by the current naming! Just to note though we are not getting MET or converting to MET. That is the input here, and then we are converting to a datetime64 based on whenever launch was as you say. I'm trying to get across we are going from MET to earth UTC times.

Reply:
To be somewhat consistent with SPICE, this could be named `met_to_et` or `met_to_j2000ns`. Ephemeris Time in spice is J2000 seconds, not j2000 nanoseconds. But in either case, this function should not be confused with a conversion to UTC.

Reply:
But we aren't doing either of those, we are converting to UTC numpy datetime64s... Do we need to wait until SPICE gets involved and just remove this function then?

Reply:
Aha, I hadn't appreciated that `datetime64` is referenced to the epoch `1970-01-01T00:00:00`.

Ultimately, yes SPICE will need to be used to convert from SHCOARSE to J2000 seconds (ET). ET is the time that SPICE requires throughout as an input to functions used to compute geometry.

ET will get converted to J2000 nanoseconds to store in CDF epoch coordinates.

I have not looked at what cdflib does to convert `datetime64` to ISTP compliant epoch in the CDF, but that is another time format translation that we need to be cognizant of. It may be that we want to skip ever translating to `datetime64` and just go from ET directly to J2000 nanoseconds for the CDF files.